hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1227,25 +1227,6 @@ def _from_type(thing: Type[Ex]) -> SearchStrategy[Ex]:
     # refactoring it's hard to do without creating circular imports.
     from hypothesis.strategies._internal import types
 
-    def as_strategy(strat_or_callable, thing):
```

Comment:
I'd prefer to avoid moving this, since we won't need it for a registered function to resolve tuples.

Reply:
Don't we still need it for tuples though? As I understand it sanity checks the user registration. Like we might take it for granted that the user registration ought to be callable, but even if it's confirmed to be callable, `as_strategy` still does other checks to ensure that the returned strategy is formed correctly, and it feels right to invoke `as_strategy` to perform those checks rather than to copy the code.

I could try leaving `as_strategy` in `core.py`, and then if a circular dependency comes up when I invoke it in `types.py`, I could try  finding some way to resolve that. I did not actually verify that this causes a circular dependency, I just assumed that it would, so I could go check that. But if I don't use `as_strategy` at all, then I guess I'd have to copy some of its checks into `types.py`, which I could do if you want but doesn't seem like the best solution to me.